schema/two_tables_many_to_many_design_recipe_template.md

```diff
@@ -0,0 +1,164 @@
+
+Remember to **always** have the primary key `id` as a first column. Its type will always be `SERIAL`.
+
+```
+# EXAMPLE:
+
+Table: items
+id: SERIAL
+name: text
+price: int
+quantity: int
+
+Table: orders
+id: SERIAL
+customer_name: text
+date: date
+```
+
+## 4. Design the Many-to-Many relationship
+
+Make sure you can answer YES to these two questions:
+
+1. Can one [TABLE ONE] have many [TABLE TWO]? (Yes/No)
+2. Can one [TABLE TWO] have many [TABLE ONE]? (Yes/No)
+
+```
+# EXAMPLE
+
+1. Can one tag have many posts? YES
+2. Can one post have many tags? YES
+```
+
+_If you would answer "No" to one of these questions, you'll probably have to implement a One-to-Many relationship, which is simpler. Use the relevant design recipe in that case._
+
+## 5. Design the Join Table
+
+The join table usually contains two columns, which are two foreign keys, each one linking to a record in the two other tables.
+
+The naming convention is `items_orders`.
+
+```
+# EXAMPLE
+
+Join table for tables: items and orders
+Join table name: items_orders
+Columns: items_id, order_id
+```
+
+## 4. Write the SQL.
+
+```sql
+-- EXAMPLE
+-- file: items_orders.sql
+
+-- Replace the table name, columm names and types.
+
+-- Create the first table.
+CREATE TABLE items (
+  id SERIAL PRIMARY KEY,
+  name text,
+  price int,
+  quantity int
+);
+
+-- Create the second table.
+CREATE TABLE orders (
+  id SERIAL PRIMARY KEY,
+  customer_name text,
+  date date
+);
+
+-- Create the join table.
+CREATE TABLE items_orders (
+  item_id int,
+  order_id int,
+  constraint fk_item foreign key(item_id) references items(id) on delete cascade,
+  constraint fk_order foreign key(order_id) references orders(id) on delete cascade,
+  PRIMARY KEY (item_id, order_id)
+);
+
+```
+
+## 5. Create the tables.
+
+```bash
+psql -h 127.0.0.1 database_name < posts_tags.sql
+```
+
+<!-- BEGIN GENERATED SECTION DO NOT EDIT -->
+
+---
+
+**How was this resource?**  
+[😫](https://airtable.com/shrUJ3t7KLMqVRFKR?prefill_Repository=makersacademy%2Fdatabases&prefill_File=resources%2Ftwo_tables_many_to_many_design_recipe_template.md&prefill_Sentiment=😫) [😕](https://airtable.com/shrUJ3t7KLMqVRFKR?prefill_Repository=makersacademy%2Fdatabases&prefill_File=resources%2Ftwo_tables_many_to_many_design_recipe_template.md&prefill_Sentiment=😕) [😐](https://airtable.com/shrUJ3t7KLMqVRFKR?prefill_Repository=makersacademy%2Fdatabases&prefill_File=resources%2Ftwo_tables_many_to_many_design_recipe_template.md&prefill_Sentiment=😐) [🙂](https://airtable.com/shrUJ3t7KLMqVRFKR?prefill_Repository=makersacademy%2Fdatabases&prefill_File=resources%2Ftwo_tables_many_to_many_design_recipe_template.md&prefill_Sentiment=🙂) [😀](https://airtable.com/shrUJ3t7KLMqVRFKR?prefill_Repository=makersacademy%2Fdatabases&prefill_File=resources%2Ftwo_tables_many_to_many_design_recipe_template.md&prefill_Sentiment=😀)  
+Click an emoji to tell us.
+
+<!-- END GENERATED SECTION DO NOT EDIT -->
```

Comment:
Great planning of your code by creating thorough recipes for your repos and tables.